frontend/src/components/PasswordChange.jsx

```diff
@@ -0,0 +1,378 @@
+                    <Form.Label>Verification Code</Form.Label>
+                    <Form.Control
+                      className={
+                        verificationCodeValidated && !verificationCodeValid
+                          ? "is-invalid"
+                          : ""
+                      }
+                      type="text"
+                      placeholder="Type your verification code"
+                      onChange={(e) => {
+                        setVerificationCode(e.target.value);
+                        isVerificationCodeValid(e.target.value);
+                        if (e.target.value.length === 0) {
+                          setVerificationCodeValidated(false);
+                        }
+                      }}
+                      value={verificationCode}
+                    />
+                    <Form.Control.Feedback type="invalid">
+                      {verificationCodeErrorMessage}
+                    </Form.Control.Feedback>
+                  </Form.Group>
+                  <Form.Group className="mb-4" controlId="formPassword">
+                    <Form.Label>Password</Form.Label>
+                    <InputGroup>
+                      <Form.Control
+                        className={
+                          passwordValidated
+                            ? passwordValid
+                              ? "is-valid"
+                              : "is-invalid"
+                            : ""
+                        }
+                        type={showPassword ? "text" : "password"}
+                        placeholder="Type your password"
+                        onChange={(e) => {
+                          setPassword(e.target.value);
+                          isPasswordSecure(e.target.value);
+                          if (e.target.value.length === 0) {
+                            setPasswordValidated(false);
+                          }
+                        }}
+                        value={password}
+                      />
+                      <InputGroup.Text
+                        className="password-icon"
+                        onClick={() => setShowPassword(!showPassword)}
+                      >
+                        {showPassword ? <EyeSlash /> : <Eye />}
+                      </InputGroup.Text>
+                      <Form.Control.Feedback type="invalid">
+                        {passwordErrorMessage}
+                      </Form.Control.Feedback>
+                    </InputGroup>
+                  </Form.Group>
+                  <Form.Group className="mb-5" controlId="formConfirmPassword">
+                    <Form.Label>Confirm Password</Form.Label>
+                    <InputGroup>
+                      <Form.Control
+                        className={
+                          confirmPasswordValidated
+                            ? passwordsMatch
+                              ? "is-valid"
```

Comment:
I think it is a good idea to have several steps to ensure a good password, such as confirming the password.